pkg/engine/validation.go

```diff
@@ -755,3 +770,31 @@ func (v *validator) substituteDeny() error {
 	v.deny = i.(*kyvernov1.Deny)
 	return nil
 }
+
+// matchesException gets excepted resources
+// checks if excepted resources include the target resource
+// return true if resources match excepted resource
+func matchesException(log logr.Logger, policyContext *PolicyContext, rule *kyvernov1.Rule, peLister kyvernov2alpha1.PolicyExceptionLister) (bool, string) {
+	exceps, err := peLister.List(labels.Everything())
+	if err != nil {
+		return false, ""
+	}
+
+	for _, v := range exceps {
+		exceptions := v.Spec.Exceptions
+
+		for _, p := range exceptions {
+			if p.PolicyName != policyContext.policy.GetName() {
+				continue
+			}
+
+			for _, r := range p.RuleNames {
+				if r == rule.Name {
+					// TODO: check if newResource is included in the excepted resources
```

Comment:
Hi folks, need some help here.
I'm wondering how we can implement the logic to check if the `exceptions.exclude` contains the `policyContext.newResource`?

Reply:
We will need to make this reusable, to evaluate if the rule matches the resource to be excluded:

https://github.com/kyverno/kyverno/blob/9d3b176defe4b7faeb6b4f664e40eef9151cb8a4/pkg/engine/utils.go#L284

Reply:
I implemented most of what you need here https://github.com/kyverno/kyverno/blob/main/cmd/cleanup-controller/handlers/cleanup/match.go (based v2beta1 types, not coupled to policy rules so easy to reuse).

It's not public yet though, we need to move that in a shared package.

Reply:
Thx @eddycharly. That helped a lot!
I move it to `pkg/utils/match.go`, so that it can be publicly be used on both sides.
Is that ok?

Reply:
@JimBugwadia I think it's a good idea to refact this part to make it reusable. Could we do it in a later PR?

Reply:
Put it in something like /pkg/utils/match/… ?
/pkg/utils is too generic IMHO

Reply:
I can move it later anyway. The tricky part here is that it depends on the api version.

Reply:
Make sense👍 Have moved it to /pkg/utils/match/..